src/routes/Groups/Groups.store.js

```diff
@@ -1,60 +1,19 @@
-import { createSelector } from 'reselect'
-
-export const MODULE_NAME = 'RelatedGroups'
-
-// Constants
-export const SET_SEARCH = `${MODULE_NAME}/SET_SEARCH`
-export const SET_SORT = `${MODULE_NAME}/SET_SORT`
-
-// Reducer
-const defaultState = {
-  sort: 'name',
-  search: ''
-}
-
-export default function reducer (state = defaultState, action) {
-  const { error, type, payload } = action
-  if (error) return state
-
-  switch (type) {
-    case SET_SEARCH:
-      return {
-        ...state,
-        search: payload
-      }
-    case SET_SORT:
-      return {
-        ...state,
-        sort: payload
-      }
-    default:
-      return state
-  }
-}
-
-export function setSearch (search) {
-  return {
-    type: SET_SEARCH,
-    payload: search
+import { createSelector as ormCreateSelector } from 'redux-orm'
+import orm from 'store/models'
+import getGroupForCurrentRoute from 'store/selectors/getGroupForCurrentRoute'
+import { getCurrentlyRelatedGroupIds } from 'store/selectors/getGroupRelationships'
+import getMyModeratedGroups from 'store/selectors/getMyModeratedGroups'
+import getRouteParam from 'store/selectors/getRouteParam'
+
+export const getPossibleRelatedGroups = ormCreateSelector(
+  orm,
+  getGroupForCurrentRoute,
+  (session, props) => getCurrentlyRelatedGroupIds(session, { groupSlug: getRouteParam('groupSlug', session, props) }),
+  getMyModeratedGroups,
+  (session, group, currentRelationships, moderatedGroups) => {
+    // TODO: check for cycles, cant add a grandparent as a child
```

Comment:
Assuming that this case isn't covered otherwise, what happens if someone does that?

Reply:
well, the main issue that might happen which i havent yet looked at is that when looking at a parent group you see all the posts of children groups you are also a member of so we would need to check for the cycle there. maybe its not actually a big deal if there are cycles... just strange and maybe indicating something weird going on